services/checks/checks_test.go

```diff
@@ -261,10 +264,140 @@ func TestFindTargets(t *testing.T) {
 		require.NoError(t, sqlDB.Close())
 	}()
 	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+	s := New(nil, nil, db)
+
+	t.Run("unknown service", func(t *testing.T) {
+		targets, err := s.findTargets(models.PostgreSQLServiceType, nil)
+		require.NoError(t, err)
+		assert.Len(t, targets, 0)
+	})
+
+	t.Run("different pmm agent versions", func(t *testing.T) {
+		node, err := models.CreateNode(db.Querier, models.GenericNodeType, &models.CreateNodeParams{
+			NodeName: "test-node",
+		})
+		require.NoError(t, err)
 
-	s := New(nil, nil, db, "2.5.0")
+		setup(t, db, "mysql1", node.NodeID, "")
+		setup(t, db, "mysql2", node.NodeID, "2.5.0")
+		setup(t, db, "mysql3", node.NodeID, "2.6.0")
+		setup(t, db, "mysql4", node.NodeID, "2.6.1")
+		setup(t, db, "mysql5", node.NodeID, "2.7.0")
+
+		tests := []struct {
+			name               string
+			minRequiredVersion *version.Parsed
+			count              int
+		}{
+			{"without version", nil, 5},
+			{"version 2.5.0", mustParseVersion("2.5.0"), 4},
+			{"version 2.6.0", mustParseVersion("2.6.0"), 3},
+			{"version 2.6.1", mustParseVersion("2.6.1"), 2},
+			{"version 2.7.0", mustParseVersion("2.7.0"), 1},
+			{"version 2.9.0", mustParseVersion("2.9.0"), 0},
+		}
+
+		for _, test := range tests {
+			test := test
+
+			t.Run(test.name, func(t *testing.T) {
+				targets, err := s.findTargets(models.MySQLServiceType, test.minRequiredVersion)
+				require.NoError(t, err)
+				assert.Len(t, targets, test.count)
+			})
+		}
+	})
+}
 
-	targets, err := s.findTargets(models.PostgreSQLServiceType)
+func setup(t *testing.T, db *reform.DB, serviceName, nodeID, pmmAgentVersion string) {
```

Comment:
Somewhat strange to see `setup` function being defined after the test that uses it.